player.py

```diff
@@ -250,7 +254,7 @@ def take_food_from_trail(self, trail, ant_positions, ant, direction):
     else:
       raise(ValueError("Direction should be 'front' or 'back'."))
 
-    while food_position in ant_positions.values():
+    while food_position in ant_positions.values() or trail[food_position] is None:
```

Comment:
A more detailed description on what foods can taken (i.e. adjacent food that is not None and does not have an ant) would be useful to have in the function header.